benchmarks_runner/eval.py

```diff
@@ -81,29 +82,31 @@ def evaluate_results(
                     message = response.get('message', None)
                     if message is None:
                         # most likely came from the ARS
-                        message = ((response.get('fields') or {}).get('data') or {}).get('message', {})
+                        message = reduce(lambda val, key: val.get(key) if val else {}, ["fields", "data", "message"], response)
         elif type(results_dir) is dict:
             if uid not in results_dir:
                 warnings.warn(f'Results for query {uid} were not found.')
             else:
                 # Grab message from dict
-                message = results_dir.get(uid, {}).get("message", {})
+                message = reduce(lambda val, key: val.get(key) if val else {}, [uid, "message"], results_dir)
 
         results_k = sorted(
-            [] if message.get('results') is None else message['results'],
-            key=lambda r: r.get('analyses', [{ 'score': 0 }])[0].get('score', 0),
+            [] if message.get('results') is None else message["results"],
+            key=lambda r: (r.get('analyses') or [{ 'score': 0 }])[0].get('score') or 0,
             reverse=True
         )[:k]
 
         if use_xref:
             # Use the biolink:xref attribute to add additional aliases to the normalizer
-            knowledge_graph = message.get("knowledge_graph", {})
-            for curie, curie_info in knowledge_graph.get('nodes', {}).items():
-                for attribute in curie_info.get('attributes', []):
-                    if attribute.get('attribute_type_id', None) != 'biolink:xref':
+            kgraph_nodes = reduce(lambda val, key: val.get(key) if val else {}, ["knowledge_graph", "nodes"], message)
```

Comment:
Same comment as above
```suggestion
            kgraph_nodes = reduce(lambda val, key: {} if val.get(key) is None else {}, ["knowledge_graph", "nodes"], message)
```